packages/next/server/on-demand-entry-handler.js

```diff
@@ -287,14 +292,27 @@ export default function onDemandEntryHandler (devMiddleware, multiCompiler, {
           if (!/^\/_next\/on-demand-entries-ping/.test(req.url)) return next()
 
           const { query } = parse(req.url, true)
-
-          if (query.page) {
-            return handlePing(query.page, res)
+          const page = query.page
+          if (!page) return next()
+
+          // Upgrade request to EventSource
+          req.socket.setKeepAlive(true)
+          res.writeHead(200, evtSourceHeaders)
+          res.write('\n')
+          clients.add(res)
```

Comment:
I wonder if we could use WeakSet for this, then you probably wouldn't have to delete on res end.

Reply:
If you don't think we need to manually close the sockets when `stop()` is called then we could get rid of the set of clients since I was just using it for that. WeakSet wouldn't work in this case if we do need to close them since WeakSet doesn't allow iterating.